### DISCUSSION TITLE
Loading the extension... Fail.

### DISCUSSION THREAD

**Discussion by Samjack1533** [wawawario2/text-generation-webui#23] [Q&A]
i would appreciate any help
![image](https://user-images.githubusercontent.com/107244975/231896609-88f2522d-e9da-42de-8ec4-9336a95a320c.png)
![image](https://user-images.githubusercontent.com/107244975/231897145-d285af2c-6858-4859-a422-9a52bbd741f5.png)

**Comment by daledavood** [comment]
I have the same problem here

**Comment by UserSpy** [comment]
I ran into a similar issue but with zarr. Took me a while, but I had to manually install the missing module into the environment. In my oobabooga I found this path "oobabooga_windows\installer_files\env". I then ran the command in a miniconda command prompt "conda activate C:\your_path\oobabooga_windows\installer_files\env" (replace "your_path" with your actual path to this, and C with whatever drive you are working off of)
Then once that activated, I typed "pip install zarr" (though you might need "pip install sentence_transformers"),

After that, I typed "conda deactivate" and then ran oobabooga with no errors.

**Reply by H2RnD** [reply]
I've fixed my issue with a workaround so I figured I'd post my workflow in case that helps anyone. (Linux Pop_OS!)

I recently upgraded my webui install and broke a few things. Long_Term_Memory was one of them. I would get the error when running start_linux.sh:

> ERROR:Failed to load the extension "long_term_memory".

ending with:

>     from sentence_transformers import SentenceTransformer
> ModuleNotFoundError: No module named sentence_transformers


First I figured I would ensure that the environment was right. I installed from the conda environment from my main directory `conda install -c conda-forge sentence-transformers`
Then I deleted long_term_memory from my extensions folder. Then I followed steps to install `git clone https://github.com/wawawario2/long_term_memory extensions/long_term_memory' to my extensions folder. 

Then 'CD ..' up to my text-generation-webui folder and gather requirements but using pip3 since I have multiple versions of Python installed `pip3 install -r extensions/long_term_memory/requirements.txt`

I ran this seeing 2 warnings for distutils classes being deprecated. `python -m pytest -v extensions/long_term_memory/`
Then I ran into an error from 'python server.py --extensions long_term_memory':

> ImportError: cannot import name GPTQConfig from transformers

I figured that transformers might need to be updated, so I ran 'pip3 install -U transformers'
I reran the python server.py command with LTM and it started without any errors. 

HOWEVER - running from start_linux.sh I still get the same error. 
So, anybody have thoughts as to why start_linux.sh gives an error loading the extension while server.py runs it without issue?
